### PR TITLE
Replace indActive for active_cells in regularizations

### PR DESCRIPTION
**Pull request by santisoler**
#### Summary

Remove the `indActive` argument in constructors of regularization classes, remove the `indActive` attribute and replace them for the supported `active_cells`. Replace the `indActive` argument in the `depth_weighting` function in favor of `active_cells`. Raise errors when those arguments are being passed and add tests checking those errors are being raised. Update code, tests and examples that were still using `indActive`.

#### PR Checklist
* [ ] If this is a work in progress PR, set as a Draft PR
* [ ] Linted my code according to the [style guides](https://docs.simpeg.xyz/content/getting_started/contributing/code-style.html).
* [ ] Added [tests](https://docs.simpeg.xyz/content/getting_started/practices.html#testing) to verify changes to the code.
* [ ] Added necessary documentation to any new functions/classes following the
      expect [style](https://docs.simpeg.xyz/content/getting_started/practices.html#documentation).
* [ ] Marked as ready for review (if this is was a draft PR), and converted
      to a Pull Request
* [ ] Tagged ``@simpeg/simpeg-developers`` when ready for review.

#### Reference issue

Part of the solution for #1302

#### Additional information

Most of these changes were cherry-picked from #1306

### CODE REVIEW

**Review by jcapriot — COMMENTED** [review]
Are you planning on grabbing the other indActive keyword arguments in the other areas in a separate PR?

**Comment by santisoler** [comment]
> Are you planning on grabbing the other indActive keyword arguments in the other areas in a separate PR?

Do you mean the `indActive` arguments in simulations and maps? If that was your question, yes. Here I just want to make the `indActive` in regularizations to raise an error, since they were already deprecated.

File: tests/base/test_regularization.py [hunk 1]
```diff
@@ -803,6 +790,56 @@ def test_weights(self, mesh):
             BaseRegularization(mesh, weights=weights, cell_weights=weights)
 
 
+class TestRemovedObjects:
+    """
+    Test if errors are raised after passing deprecated arguments or trying to
```
Comment (santisoler) [inline tests/base/test_regularization.py:795]:
```suggestion
    Test if errors are raised after passing removed arguments or trying to
```

**Comment by jcapriot** [comment]
> > Are you planning on grabbing the other indActive keyword arguments in the other areas in a separate PR?
> 
> Do you mean the `indActive` arguments in simulations and maps? If that was your question, yes. Here I just want to make the `indActive` in regularizations to raise an error, since they were already deprecated.

Yeah that's what I was referring to. Just double checking as I'd like to make this change everywhere for the next release (so it's not mixed between the two different conventions).

**Review by jcapriot — APPROVED** [review]

**Pull request merged**